src/windows.rs

```diff
@@ -27,31 +42,32 @@ impl ProcessHandleExt for ProcessHandle {
 }
 
 /// A `Pid` can be turned into a `ProcessHandle` with `OpenProcess`.
-impl TryIntoProcessHandle for minwindef::DWORD {
+impl TryIntoProcessHandle for windows::DWORD {
     fn try_into_process_handle(&self) -> std::io::Result<ProcessHandle> {
-        let handle = unsafe {
-            winapi::um::processthreadsapi::OpenProcess(
-                winapi::um::winnt::PROCESS_CREATE_THREAD
-                    | winapi::um::winnt::PROCESS_QUERY_INFORMATION
-                    | winapi::um::winnt::PROCESS_VM_READ
-                    | winapi::um::winnt::PROCESS_VM_WRITE
-                    | winapi::um::winnt::PROCESS_VM_OPERATION,
-                winapi::shared::minwindef::FALSE,
-                *self,
-            )
-        };
-        if handle == (0 as winapi::um::winnt::HANDLE) {
-            Err(std::io::Error::last_os_error())
-        } else {
-            Ok((handle, Architecture::from_native()))
-        }
+        Ok((
+            unsafe {
+                windows::OpenProcess(
+                    windows::PROCESS_CREATE_THREAD
+                        | windows::PROCESS_QUERY_INFORMATION
+                        | windows::PROCESS_VM_READ
+                        | windows::PROCESS_VM_WRITE
+                        | windows::PROCESS_VM_OPERATION,
+                    false,
+                    *self,
+                )
+            }?,
+            Architecture::from_native(),
+        ))
     }
 }
 
 /// A `std::process::Child` has a `HANDLE` from calling `CreateProcess`.
 impl TryIntoProcessHandle for Child {
     fn try_into_process_handle(&self) -> std::io::Result<ProcessHandle> {
-        Ok((self.as_raw_handle().cast(), Architecture::from_native()))
+        Ok((
+            windows::HANDLE(self.as_raw_handle() as _),
```

Comment:
Could you check that `cargo clippy` passes with this change? It tends to strongly prefer `.cast()` to `as _`, but it may be fine if `windows::HANDLE` isn't a pointer type.

Reply:
clippy gives no warnings for me here.

Reply:
I changed all target types of `as`-casts in windows.rs to be explicit, even though clippy didn't complain about them.